propagation/trace_context.go

```diff
@@ -155,6 +146,29 @@ func (tc TraceContext) extract(carrier TextMapCarrier) trace.SpanContext {
 	return sc
 }
 
+// upperHex detect hex is upper case.
+func upperHex(v string) bool {
+	for _, c := range v {
+		if c >= 'A' && c <= 'F' {
+			return true
+		}
+	}
+	return false
+}
+
+func extractPart(dst []byte, h *string, n int) bool {
+	part, left, _ := strings.Cut(*h, parentDelimiter)
```

Comment:
If `Cut` does not find `parentDelimiter` it will return `"h"`, `""`, `false`. Shouldn't we check the returned `found` value before setting `h` below?

Reply:
```
part, left, found := strings.Cut(*h, sep)
if (!found) {
  *h = ""
}
```

equals to 

```
part, left, _ := strings.Cut(*h, sep)
*h = left (left == "")
```